.github/workflows/main.yml

```
@@ -12,13 +12,24 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - name: Setup Go
-        uses: actions/setup-go@v3.2.0
+        uses: actions/setup-go@v4
         with:
           go-version: ${{ matrix.go_version }}
-      - run: ./.ci.gogenerate.sh
-      - run: ./.ci.gofmt.sh
-      - run: ./.ci.govet.sh
-      - run: go test -v -race ./...
+      - name: Check go mod tidy
+        run: |
+          go mod tidy
+          (cd _codegen && go mod tidy)
+          git --no-pager diff && [[ -z $(git status -s) ]]
```

We can use [`--exit-code`](https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---exit-code) parameter here and below:
```suggestion
          git --no-pager diff --exit-code
```

---

That wouldn't handle the case where there are new files generated unfortunately., which is what `git status -s` catches.